### DISCUSSION TITLE
Executing repository.flush() and executing .toObject() takes 5s to complete

### DISCUSSION THREAD

**Discussion by Virtouzo** [mikro-orm/mikro-orm#4163] [Q&A]
I didn't this open this issue as a bug, since I am not sure if what I am about to describe is a bug.

I am doing the following process:

```js
    const order = await orderRepo.findOneOrFail(
        { ID: orderID },
        { populate: ["invitations", "invitations.service"] }
    );

    const newInvitations = invitations.map((invitationDTO) => {
        const service = serviceRepo.getReference(invitationDTO.serviceID);
        const invitation = participationRepo.create({
            service,
            primary: invitationDTO.primary,
            estimatedCost: invitationDTO.estimatedCost,
            order,
            expiresAt: invitationDTO.expiresAt,
        });

        return invitation;
    });
    await participationRepo.persistAndFlush(newInvitations);
    order.servicesJoinTimeoutAt = invitations[0].expiresAt;

    const newInvs = newInvitations.map((inv) => {
        console.time('toObject')
        const object = wrap(inv).toObject();
        console.timeEnd('toObject')
        return object;
    });
```

the `toObject` timer takes around 5s if I call `participationRepo.persistAndFlush(newInvitations);` and it takes around ~50ms if I change it to `participationRepo.persist(newInvitations);`. 

Note: the `service` and `order` entities both have `@ManyToOne` relationship on the `ParticipationEntity`.

Is there some kind of bug going on or am I doing something wrong? What could be the possible reasons for the `.toObject` taking so long, considering that I am only testing this just with a single entity.

**Comment by B4nan** [comment]
> considering that I am only testing this just with a single entity.

It's probably not just "a single entity", all the relations are entities as well. What do you get if you `console.log` that single entity?

> the toObject timer takes around 5s if I call participationRepo.persistAndFlush(newInvitations); and it takes around ~50ms if I change it to participationRepo.persist(newInvitations);.

By flushing, all the new entities will be marked as populated, which affects the serialization, so it might be about that. Maybe you could try [explicit serialization](https://mikro-orm.io/docs/serializing#explicit-serialization).

Will be happy to look at a repro if you can provide one, maybe there is something to improve.

**Reply by Virtouzo** [reply]
Just wanted to make a follow-up here, that explicit serialization solved the problem. It's much faster.

Sorry for such a long delayed follow-up.
